tests/test_core.py

```diff
@@ -23,15 +24,112 @@ class test_PairMatrix(unittest.TestCase):
+        self.assertEqual(m.types, pair_types)
+        self.assertEqual(m._data, pair_dict)
+
+        #test construction with list input
+        m = relentless.core.PairMatrix(types=['A','B'])
+        self.assertEqual(m.types, pair_types)
+        self.assertEqual(m._data, pair_dict)
+
+        #test construction with numpy array input
+        m = relentless.core.PairMatrix(types=np.array(['A','B']))
+        self.assertEqual(m.types, pair_types)
+        self.assertEqual(m._data, pair_dict)
+
+        pair_types = ('A',)
+        pair_dict = {('A','A'):{}}
+
+        #test construction with single type tuple
+        m = relentless.core.PairMatrix(types=('A',))
+        self.assertEqual(m.types, pair_types)
+        self.assertEqual(m._data, pair_dict)
 
     def test_accessors(self):
         """Test get and set methods on pairs."""
-        raise NotImplementedError()
+
+        m = relentless.core.PairMatrix(types=('A','B'))
+
+        #test set and get for each pair type
+        m['A','A']['energy'] = 1.0
+        self.assertEqual(m['A','A']['energy'], 1.0)
+        self.assertEqual(m['A','B'], {})
+        self.assertEqual(m['B','B'], {})
+
+        m['A','B']['energy'] = -1.0
+        self.assertEqual(m['A','A']['energy'], 1.0)
+        self.assertEqual(m['A','B']['energy'], -1.0)
+        self.assertEqual(m['B','B'], {})
+
+        m['B','B']['energy'] = 1.0
+        self.assertEqual(m['A','A']['energy'], 1.0)
+        self.assertEqual(m['A','B']['energy'], -1.0)
+        self.assertEqual(m['B','B']['energy'], 1.0)
+
+        #test re-set and get
+        m['A','A']['energy'] = 2.0
+        self.assertEqual(m['A','A']['energy'], 2.0)
+        self.assertEqual(m['A','B']['energy'], -1.0)
+        self.assertEqual(m['B','B']['energy'], 1.0)
+
+        m['A','B']['energy'] = -1.5
+        self.assertEqual(m['A','A']['energy'], 2.0)
+        self.assertEqual(m['A','B']['energy'], -1.5)
+        self.assertEqual(m['B','B']['energy'], 1.0)
+
+        m['B','B']['energy'] = 0.0
+        self.assertEqual(m['A','A']['energy'], 2.0)
+        self.assertEqual(m['A','B']['energy'], -1.5)
+        self.assertEqual(m['B','B']['energy'], 0.0)
+
+        #test setting multiple parameters and get
+        m['A','A']['mass'] = 1.0
+        self.assertEqual(m['A','A']['mass'], 1.0)
+        self.assertEqual(m['A','A']['energy'], 2.0)
+        self.assertEqual(m['A','A'], {'energy':2.0, 'mass':1.0})
+
+        m['A','B']['mass'] = 3.0
+        self.assertEqual(m['A','B']['mass'], 3.0)
+        self.assertEqual(m['A','B']['energy'], -1.5)
+        self.assertEqual(m['A','B'], {'energy':-1.5, 'mass':3.0})
+
+        m['B','B']['mass'] = 5.0
+        self.assertEqual(m['B','B']['mass'], 5.0)
+        self.assertEqual(m['B','B']['energy'], 0.0)
+        self.assertEqual(m['B','B'], {'energy':0.0, 'mass':5.0})
+
+        m_real = {('A','B'):{'energy':-1.5, 'mass':3.0}, ('A','A'):{'energy':2.0, 'mass':1.0}, ('B','B'):{'energy':0.0, 'mass':5.0}}
```

Comment:
Your tests above are sufficient, you can remove this check---again, because it is looking at an internal "private" variable, beginning with `_`, so it should be allowed to be silently changeable.